dev_scripts/inventory.py

```diff
@@ -0,0 +1,793 @@
+        except Exception as e:
+            raise InvException(f"Error extracting zip archive: {e}") from e
+    else:
+        raise InvException(f"Unsupported archive type for file {archive_path}")
+
+    if flatten:
+        flatten_extracted_files(destination)
+
+    logger.debug(f"Successfully extracted '{archive_path}'")
+
+
+def get_platform_assets(assets, platform):
+    """
+    List the assets that are associated with a specific platform.
+    """
+
+    plat_assets = {}
+    for asset_name, asset_entry in assets.items():
+        if platform in asset_entry:
+            plat_assets[asset_name] = asset_entry[platform]
+        elif "all" in asset_entry:
+            plat_assets[asset_name] = asset_entry["all"]
+    return plat_assets
+
+
+def chmod_exec(path):
+    if path.is_dir():
+        for root, _, files in path.walk():
+            for name in files:
+                f = root / name
+                f.chmod(f.stat().st_mode | 0o111)
+    else:
+        path.chmod(path.stat().st_mode | 0o111)
+
+
+def compute_asset_lock(asset_name, asset):
+    try:
+        repo = asset["repo"]
+        version_range = asset["version"]
+        asset_map = asset["platform"]  # mapping platform -> asset file name
+        destination_str = asset["destination"]
+        executable = asset.get("executable", False)
+        extract = asset.get("extract", False)
+    except KeyError as e:
+        raise InvException(f"Required field {e} is missing")
+
+    if extract:
+        extract = determine_extract_opts(extract)
+
+    logger.debug(
+        f"Fetching a release that satisfies version range '{version_range}' for repo"
+        f" '{repo}'"
+    )
+    release = get_latest_release(repo, version_range)
+    version = release["tag_name"].lstrip("v")
+    logger.debug(f"Found release '{version}' for repo '{repo}'")
+
+    asset_lock_data = {}
+    # Process each defined platform key in the asset_map
+    for plat_key, plat_name in asset_map.items():
+        logger.debug(f"Getting download URL for asset '{asset_name}' of repo '{repo}'")
+        download_url = get_download_url(release, plat_name)
+        logger.debug(f"Found download URL: {download_url}")
+
+        if extract:
+            extract = extract.copy()
+            extract["filetype"] = detect_archive_type(plat_name)
+
+        logger.info(
+            f"Hashing asset '{asset_name}' of repo '{repo}' for platform"
+            f" '{plat_key}'..."
+        )
+        checksum = hash_asset(download_url)
+        logger.debug(f"Computed the following SHA-256 checksum: {checksum}")
+        asset_lock_data[plat_key] = {
+            "repo": repo,
+            "download_url": download_url,
+            "version": version,
+            "checksum": checksum,
+            "executable": executable,
+            "destination": destination_str,
+            "extract": extract,
+        }
+
+    return asset_lock_data
+
+
+def sync_asset(asset_name, target_plat, asset):
```

Comment:
Proposal to use a different naming, like this:
```suggestion
def sync_asset(name, platform, asset_dict):
```

Reply:
Done in f9580525f8734e28871c2e816dea6d43c64857d7.